packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -82,25 +67,27 @@ class DatadogRumPlugin internal constructor(
     private lateinit var channel: MethodChannel
     private lateinit var binding: FlutterPlugin.FlutterPluginBinding
 
-    var rum: RumMonitor? = rumInstance
+    var rum: RumMonitor? = null
         private set
 
-    val mapperPerf = PerformanceTracker()
-    val mapperPerfMainThread = PerformanceTracker()
-    var mapperTimeouts = 0
-
     // Might need a better way to deal with this. There's weird shared responsibility for
     // telemetry between the core and RUM.
     var telemetryOverrides: DatadogSdkPlugin.ConfigurationTelemetryOverrides? = null
 
     fun attachToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
         channel = MethodChannel(flutterPluginBinding.binaryMessenger, "datadog_sdk_flutter.rum")
         channel.setMethodCallHandler(this)
+        eventMapper.addChannel(channel)
 
         binding = flutterPluginBinding
+
+        if (GlobalRumMonitor.isRegistered()) {
+            rum = GlobalRumMonitor.get()
+        }
```

Comment:
why it is needed to keep the reference, when we can simply call `GlobalRumMonitor.get()` each time?

Reply:
It's not now, but it's a smaller change to keep the reference for this PR. Do you think it's an issue to keep it the way it is?